### DISCUSSION TITLE
why cookies are using uuids which is not cryptographically secure?

### DISCUSSION THREAD

**Discussion by HosMercury** [maxcountryman/tower-sessions#172] [Q&A] [answered]
title

**Comment by maxcountryman** [comment]
Session IDs are [generated using the rand crate](https://github.com/maxcountryman/tower-sessions/blob/d744dc833679f66568ca77f606d596aa5f960775/tower-sessions-core/src/session.rs#L868).

**Reply by HosMercury** [reply]
But afaik threadRng is not crypt secure ..
.... OaRng is ...

**Reply by maxcountryman** [reply, marked as answer]
[ThreadRng](https://docs.rs/rand/latest/rand/rngs/struct.ThreadRng.html) implements [CryptoRng](https://docs.rs/rand/latest/rand/trait.CryptoRng.html).

See [StdRng](https://docs.rs/rand/latest/rand/rngs/struct.StdRng.html) regarding its secure nature (ChaCha is the current cipher).
